pallets/nfts/src/features/approvals.rs

```diff
@@ -122,12 +122,19 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			}
 		}
 
+		// Cannot revoke approval for a specific collection item if the delegate has
+		// permission to transfer all collection items owned by the origin.
+		ensure!(
+			!CollectionApprovals::<T, I>::contains_key((collection, &details.owner, &delegate)),
+			Error::<T, I>::DelegateApprovalConflict
```

Comment:
Not tested in `cancel_approval`

Reply:
Please ensure from now on that every error of a dispatchable is tested. These mistakes happen too often. 

If we change a dispatchable we only want to test the newly added logic.

Reply:
Resolved this by improving test cases of `cancel_approval_works`, `cancel_approval_work_with_admin` and `cancel_approval_work_with_force`